src/execution.jl

```diff
@@ -29,15 +29,15 @@ end
 # execution #
 #############
 # Note that trials executed via `run` and `lineartrial` are always executed at the top-level
-# scope of the module returned by `current_module()`. This is to avoid any weird quirks
+# scope of the module returned by `@__MODULE__`. This is to avoid any weird quirks
 # when calling the experiment from within other contexts.
 
 sample(b::Benchmark, args...) = error("no execution method defined on type $(typeof(b))")
 _run(b::Benchmark, args...; kwargs...) = error("no execution method defined on type $(typeof(b))")
 
 # return (Trial, result) tuple, where result is the result of the benchmarked expression
 function run_result(b::Benchmark, p::Parameters = b.params; kwargs...)
-    return eval(current_module(), :(BenchmarkTools._run($(b), $(p); $(kwargs...))))
+    return eval(@__MODULE__, :(BenchmarkTools._run($(b), $(p); $(kwargs...))))
```

Comment:
The PR's intent is to remove the dynamic scope semantics so it is not possible to implement `current_module()` anymore. If you have a generic function call, it will be impossible for it to know from where it is called so such information has to be passed down explicitly.